### PR TITLE
Try to persist the Pane's working directory if we know what it is.

### PR DESCRIPTION
**Pull request by Rosefield**
## Summary of the Pull Request
Try to save the working directory if we know what it is (just copied what was done in duplicating a pane). I overlooked this in my original implementation that always used the settings StartingDirectory.

 
## References
#9800 

## PR Checklist
* [ ] Closes #xxx
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [ ] Tests added/passed
* [ ] Documentation updated. If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/terminal) and link it here: #xxx
* [ ] Schema updated.
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Validation Steps Performed
Tried setting the working directory using the OSC 9;9 escape and confirmed that the directory saves correctly.

### CODE REVIEW

**Review by zadjii-msft — APPROVED** [review]
this is a really cool idea, I love it

File: src/cascadia/TerminalApp/Pane.cpp [hunk 1]
```diff
@@ -128,7 +128,15 @@ NewTerminalArgs Pane::GetTerminalArgsForPane() const
     auto controlSettings = _control.Settings().as<TerminalSettings>();
 
     args.Profile(controlSettings.ProfileName());
-    args.StartingDirectory(controlSettings.StartingDirectory());
+    // If we know the user's working directory use it instead of the profile.
+    if (!_control.WorkingDirectory().empty())
+    {
+        args.StartingDirectory(_control.WorkingDirectory());
+    }
```
Comment (lhecker) [inline src/cascadia/TerminalApp/Pane.cpp:135]: You can prevent the double-COM-call with
```suggestion
    if (const auto path = _control.WorkingDirectory(); !path.empty())
    {
        args.StartingDirectory(path);
    }
```

I love the syntax... it strongly reminds me of Go.

**Review by lhecker — APPROVED** [review]

**Comment by lhecker** [comment]
That commit message made me genuinely laugh aloud. 😄

To be fair in Go their use is a bit more prevalent as the language lacks sum types (just like C++, but their tuples are way more ergonomic, so...).
So you often see tuples being returned of the `(T, bool)` (aka `std::pair<T, bool>`) kind and you get code like this:
```go
if value, found := myHashMap[foo]; found {
    ...
}
```

I still find it useful in C++ though, since you never know if a getter is a trivial or - for the lack of better words - a "computing" one and I'm trying to proactively avoid falling into the latter ones. Please feel free to disagree in reviews though. :)

**Comment by Rosefield** [comment]
I was completely ok with the `operator bool` things since I'm so used to writing `if let Some(x) = foo()` in rust. Rust recently approved  `if let` chains as a feature so I'll have to adapt to the new landscape regardless. Its all just sugar around scoping anyways 🤷

**Pull request merged**